client/setupTests.ts

```diff
@@ -2,8 +2,17 @@
 import '@testing-library/jest-dom';
 import '~/i18n/i18n';
 
+import nodeFetch, { Request, Response } from 'node-fetch';
+
 import { server } from './src/mocks/server';
 
+// Monkey-patch fetch to allow tests to work in node (there's currently an
+// issue with the node implementation of fetch in Undici, see
+// https://github.com/apollographql/apollo-client/issues/10785#issuecomment-1527830658)
+global.fetch = nodeFetch;
+global.Request = Request;
+global.Response = Response;
```

Comment:
I don't see this as the solution, but there again, you don't explain what the problem is your seeing. I can't say I've had any trouble with fetch, it is no longer experimental in the version of NodeJS we are using either. I do admit, I've not cloned this to run the tests with/without this though, but without an explanation to what it is fixing, I'm dubious of the solution.

The link you refer to is do do with apollo graphql, something we don't use, so why should that be listened too? Having read the comment, it is talking about NextJS undici, again something we are not using, so I don't see any of this as being relevant.

I have had no trouble at all with node and fetch, been using it all weekend writing my own versions of these tests and had no trouble.

Another issue I have with this solution is that we don't directly have `node-fetch` as a dependency, so it is just a quirk of some other dependency that it even exists in our setup, that makes this solution very brittle, if that dependency were updated/removed, this too breaks.